peer.go

```diff
@@ -116,11 +116,12 @@ func (l *PeerList) Add(hostPort string) *Peer {
 	return p
 }
 
-// Get returns a peer from the peer list, or nil if none can be found.
-func (l *PeerList) Get(prevSelected map[string]struct{}) (*Peer, error) {
+// GetNew returns a new, previously un-selected peer from the peer list, or nil,
+// if no new un-selected peer can be found.
```

Comment:
instead of returning `nil, nil,` I think we should return `nil, ErrNoNewPeers` or something